### PR TITLE
Some progress on the plane

### PR DESCRIPTION
**Pull request by pberkes**
Some work on VersionName

Most important thing is a proposal for an implementation of RunVersionName

Otherwise
1) merged the datastore implementation
2) made the VersionName subclasses a bit more pythonic

### CODE REVIEW

File: pond/version_name.py [hunk 1]
```diff
@@ -34,16 +35,17 @@ def parse(klass, version_name: str) -> 'VersionName':
...
             If the version name cannot be parsed
         """
-        # TODO: this part should be refactored to support any subclass of VersionName.
-        #       However, for now, this ugly code will do the job :)
-        methods = [SimpleVersionName.parse]
-        version = None
-        for method in methods:
+        # Only first-level subclasses for the moment, it should be sufficient
+        # At the same time, we give up defining a version name priority, and will return the
+        # first VersionName subclass that can parse the string
```
Comment (smrtl) [inline pond/version_name.py:40]: Should be fine, we shouldn't have collisions in the format

File: pond/version_name.py [hunk 2]
```diff
@@ -82,98 +84,119 @@ def __repr__(self) -> str:
...
-            raise InvalidVersionName(version_name)
 
-    def next(self) -> VersionName:
-        return DateVersionName(self.dt + timedelta(days=1))
+class RunVersionName(VersionName):
+    """Run version names are composed by a run ID and a version number."""
 
-    def __str__(self) -> str:
-        return self.dt.date().isoformat()
+    _FORMAT = re.compile('^run_([A-Za-z0-9_]*)_v?([1-9][0-9]*)$')
```
Comment (smrtl) [inline pond/version_name.py:165]: Then we loose the ability to use any other kind of version name beside simple version names... But might be sufficient for now... And actually we could subclass `RunVersionName` if needed

**Review by smrtl — APPROVED** [review]
👍

File: pond/version_name.py (hunk 2 above)
Comment (pberkes) [reply pond/version_name.py:165]: that's what I thought, YAGNI

**Pull request merged**